src/plugins/cams/datasource.py

```
@@ -0,0 +1,509 @@
+                    city="",
+                    country="",
+                    metadata={
+                        'data_source': 'cams',
+                        'type': 'grid_point',
+                        'resolution': f'{self.resolution}x{self.resolution} degrees'
+                    }
+                )
+                locations.append(location)
+                grid_count += 1
+                
+        logger.info(f"Generated {len(locations)} CAMS locations ({len(japan_cities)} cities, {grid_count} grid points)")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)